libs/data/bbhnet/data/transforms/whitening.py

```diff
@@ -88,17 +113,75 @@ def forward(self, X: torch.Tensor) -> torch.Tensor:
         X = X.transpose(2, 0)
         X = X - X.mean(axis=0)
         X = X.transpose(0, 2)
-        X *= self.window
+
+        self.pad = int(np.ceil(self.time_domain_filter.size(-1) / 2))

```

Comment:
Probably worth either assigning as an attribute up front, or just using as a variable in this namespace (i.e. no `self.`, just `pad`)